cypress/integration/core_opensearch_dashboards/opensearch_dashboards/apps/query_enhancements/autocomplete_ui.spec.js

```diff
@@ -0,0 +1,151 @@
+              });
+            }
+            cy.navigateToWorkSpaceSpecificPage({
+              workspaceName: workspaceName,
+              page: 'discover',
+              isEnhancement: true,
+            });
+          });
+
+          it('should verify suggestion widget and its hint', () => {
+            // Setup
+            cy.setDataset(config.dataset, DATASOURCE_NAME, config.datasetType);
+            cy.setQueryLanguage(config.language);
+            setDatePickerDatesAndSearchIfRelevant(config.language);
+            cy.clearQueryEditor();
+            const editorType =
+              config.language === QueryLanguages.DQL.name
+                ? 'osdQueryEditor__singleLine'
+                : 'osdQueryEditor__multiLine';
+
+            cy.getElementByTestId(editorType)
+              .find('.monaco-editor')
+              .should('be.visible')
+              .within(() => {
+                // Show suggestion and hint with retry
+                showSuggestionAndHint();
+
+                // Verify suggestions are visible
+                cy.get('.monaco-list-row').should('be.visible').should('have.length.at.least', 1);
+
+                // Sends ESC and verifies widgets are hidden
+                hideWidgets();
+
+                // TODO: Add test for having another focused window after bug is fixed
+                // https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8973
+              });
+          });
+
+          it('should build query using mouse interactions', () => {
+            // Setup
+            cy.setDataset(config.dataset, DATASOURCE_NAME, config.datasetType);
+            cy.setQueryLanguage(config.language);
+            setDatePickerDatesAndSearchIfRelevant(config.language);
+            cy.clearQueryEditor();
+
+            createQuery(config, false); // use mouse
+
+            // Run with mouse click
+            cy.getElementByTestId('querySubmitButton').click();
+
+            cy.waitForLoader(true);
+            cy.wait(1000);
+            validateQueryResults('unique_category', 'Configuration');
+          });
+
+          it('should build query using keyboard shortcuts', () => {
+            cy.setDataset(config.dataset, DATASOURCE_NAME, config.datasetType);
+            cy.setQueryLanguage(config.language);
+            setDatePickerDatesAndSearchIfRelevant(config.language);
+            cy.clearQueryEditor();
+
+            createQuery(config, true); // use keyboard
+
+            // Run with keyboard shortcut
+            if (config.language === QueryLanguages.DQL.name) {
+              cy.get('.inputarea').type('{enter}');
+            } else {
+              // SQL and PPL should use cy.get('.inputarea').type('{cmd+enter}')
+              // But it is not working in Remote CI
+              // TODO: investigate and fix
+              cy.getElementByTestId('querySubmitButton').click();
+            }
+
+            cy.waitForLoader(true);
+            cy.wait(2000);
```

Comment:
Noticed that we are using `cy.wait()` for some time after we use the `cy.waitForLoader()` function in a few cases. Is there a reason why we need the extra wait?

Reply:
Just see some flakiness when test 3 times in a row. Put a bit more time can ensure they success 3 times.